### PR TITLE
Optimize the authentication code

### PR DESCRIPTION
**Pull request by luobily**
The commit can make the code's logic more clear

### CODE REVIEW

**Comment by sttts** [comment]
@k8s-bot ok to test

**Comment by sttts** [comment]
@luobily thanks for your contribution!

For accepting the code, please sign the CNCF CLA: https://github.com/kubernetes/kubernetes/wiki/CLA-FAQ

File: pkg/kubeapiserver/authenticator/config.go [hunk 1]
```diff
@@ -196,14 +196,11 @@ func (config AuthenticatorConfig) New() (authenticator.Request, *spec.SecurityDe
 	if len(authenticators) == 0 {
 		if config.Anonymous {
 			return anonymous.NewAuthenticator(), &securityDefinitions, nil
+		} else {
```
Comment (deads2k) [inline pkg/kubeapiserver/authenticator/config.go]: no else is needed here and the golang guidance is avoid it.

**Comment by deads2k** [comment]
one comment, squash, and then lgtm.

**Pull request closed without merging**